v21.2/change-data-capture-overview.md

```diff
@@ -23,6 +23,26 @@ The main feature of CDC is the changefeed, which targets an allowlist of tables,
 
 See [Ordering Guarantees](use-changefeeds.html#ordering-guarantees) for detail on CockroachDB's at-least-once-delivery-guarantee as well as explanation on how rows are emitted.
 
+## How does a changefeed work?
+
+When a changefeed is started on a node, that node becomes the aggregator node for the changefeed job (CRDB Node 2 in the diagram). The aggregator node acts as an administrator: keeping track of all other nodes during job execution and the changefeed work as it completes. In the unlikely event that the changefeed’s aggregator node were to fail during the job, the aggregator role will move to a different node and the changefeed will restart from the last checkpoint.
+
+The changefeed job will run across nodes in the cluster to access changed data in the watched table. Typically, the [leaseholder](architecture/replication-layer.html#leases) for a particular range (or the range’s replica) determines which node emits the changefeed data.
+
+As each node is working, it sends back checkpoint progress to the aggregator node that gathers this information to update the high-water mark timestamp. The high-water mark acts as a checkpoint for the changefeed’s job progress, and guarantees that all changes before (or at) the timestamp have been emitted. If restarted, the changefeed will send duplicate messages from the high-water mark timestamp. For more detail on when messages are emitted, see [Ordering Guarantees](use-changefeeds.html#ordering-guarantees).
+
+<img src="{{ 'images/v21.2/changefeed-structure.png' | relative_url }}" alt="Changefeed process in a 3-node cluster" style="max-width:100%" />
+
+With the [`resolved` option](create-changefeed.html#resolved-option) specified when a changefeed is started, the aggregator node will send the resolved timestamp (i.e., the high-water mark) to each endpoint in the sink. For example, when using [Kafka](changefeed-sinks.html#kafka) this will be sent as a message to each partition; for [cloud storage](changefeed-sinks.html#cloud-storage-sink), this will be emitted as a resolved timestamp file.
```

Comment:
The resolved option, btw, is the only mechanism how core style changefeeds would know how far along they are if an error occurred.